rust-query-crlite/src/main.rs

```diff
@@ -0,0 +1,794 @@
+        }
+        let mut coverage = HashMap::new();
+        for _ in 0..count {
+            let mut coverage_entry = [0u8; COVERAGE_V1_ENTRY_BYTES];
+            match reader.read_exact(&mut coverage_entry) {
+                Ok(()) => (),
+                _ => return Err(CRLiteDBError::from("truncated CRLite coverage file")),
+            };
+            let log_id = &coverage_entry[0..32];
+            let min_timestamp = match (&coverage_entry[32..40]).read_u64::<LittleEndian>() {
+                Ok(value) => value,
+                _ => return Err(CRLiteDBError::from("truncated CRLite coverage file")),
+            };
+            let max_timestamp = match (&coverage_entry[40..48]).read_u64::<LittleEndian>() {
+                Ok(value) => value,
+                _ => return Err(CRLiteDBError::from("truncated CRLite coverage file")),
+            };
+            coverage.insert(log_id.to_vec(), (min_timestamp, max_timestamp));
+        }
+        Ok(Coverage(coverage))
+    }
+
+    fn contains(&self, cert: &X509Certificate) -> bool {
+        let sct_extension = match cert.tbs_certificate.get_extension_unique(OID_SCT_EXTENSION) {
+            Ok(Some(sct_extension)) => sct_extension,
+            _ => return false,
+        };
+        let scts = match sct_extension.parsed_extension() {
+            ParsedExtension::SCT(scts) => scts,
+            _ => return false,
+        };
+        for sct in scts.iter() {
+            if let Some((min, max)) = self.0.get(sct.id.key_id.as_ref()) {
+                if *min <= sct.timestamp && sct.timestamp <= *max {
+                    debug!(
+                        "Logged at {} by enrolled log {}",
+                        sct.timestamp,
+                        base64::encode(sct.id.key_id)
+                    );
+                    return true;
+                }
+            }
+        }
+        false
+    }
+}
+
+struct Enrollment(HashSet<EnrollmentKey>);
+impl Enrollment {
+    fn from_bytes(bytes: &[u8]) -> Result<Self, CRLiteDBError> {
+        let mut enrollment = HashSet::new();
+        let mut reader = BufReader::new(bytes);
+        match reader.read_u8() {
+            Ok(ENROLLMENT_SERIALIZATION_VERSION) => (),
+            _ => return Err(CRLiteDBError::from("unknown CRLite enrollment version")),
+        }
+        if (bytes.len() - 1) % ENROLLMENT_V1_ENTRY_BYTES != 0 {
+            return Err(CRLiteDBError::from("truncted CRLite enrollment file"));
+        }
+        let enrollment_count = (bytes.len() - 1) / ENROLLMENT_V1_ENTRY_BYTES;
+        for _ in 0..enrollment_count {
+            let mut enrollment_entry = [0u8; ENROLLMENT_V1_ENTRY_BYTES];
+            match reader.read_exact(&mut enrollment_entry) {
+                Ok(()) => (),
+                _ => return Err(CRLiteDBError::from("truncted CRLite enrollment file")),
+            };
+            let issuer_id = &enrollment_entry[..];
+            enrollment.insert(issuer_id.to_vec());
```

Comment:
We can avoid `issuer_id` altogether and just say `enrollment_entry.to_vec()` I think.